Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -396,7 +396,7 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable, IBufferPr
             return cast(format, null);
         }
 
-        public MemoryView cast([NotNull]string format, [NotNull]object shape) {
+        public MemoryView cast([NotNull]string format, [NotNull, AllowNull]object shape) {
```

Comment:
`[NotNull, AllowNull]` — I find it not very readable. Not to say the confusion with `NotNull` from `System.Diagnostics.CodeAnalysis`. How about adopting a convention to always `using`-renaming `Microsoft.Scripting.Runtime.NotNullAttribute` to `NotNoneAttribute`?

Reply:
I would be fine with `NotNone` if there is a conflict.
```C#
using NotNoneAttribute = Microsoft.Scripting.Runtime.NotNullAttribute;
```